### PR TITLE
Add tr gwas

### PR DESCRIPTION
**Pull request by nicholema**
add tr-gwas script and wdl. It enables running tr gwas genome wide on AoU cloud environment

### CODE REVIEW

File: tr-gwas/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+## Detailed usage
+
+Required arguments:
+
+* `--name <STR>`: name of the run
+
+Optional arguments:
+
+* `--phenotype <STR>`: name of targeted phenotype, seperated by comma. Default: None, run all 
+* `--cohort <STR>`: name of targeted cohort, seperated by comma.  Options: AFR, EUR, NOT_AFR, ALL . Default: None, run all
```
Comment (gymreklab) [inline tr-gwas/README.md:71]: is there a place where we describe how these cohorts were created? maybe in a README from Tara's work on preprocessing? would be good to link users there so could see a description of what these cohorts are and how many samples are in each

File: tr-gwas/tr_gwas.wdl [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            }
+    }   
+    
+
+    output {
+        Array[Array[File]] gwas_outputs = run_tr_gwas.outfile
+    }
+
+    meta {
+        description: "Use plink2 to run genome-wide TR GWAS. It runs GWAS on each chromosome for each phenotype and merges summary statstics across all chromsomes."
```
Comment (gymreklab) [inline tr-gwas/tr_gwas.wdl:49]: minor typo: chromsomes -> chromosomes

File: tr-gwas/tr_gwas.wdl [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            gwas_outfiles+="~{out_prefix}_${chrom_outprefix}_~{sample_name}.phenotype.glm.linear "
+
+        done
+        
+        # Concatenate all results
+        head -n 1 ${gwas_outfiles} > "~{out_prefix}_~{sample_name}_gwas.tab"
+
+        for file in ${gwas_outfiles}
+        do
+            grep -v POS "$file" >> "~{out_prefix}_~{sample_name}_gwas.tab"
```
Comment (gymreklab) [inline tr-gwas/tr_gwas.wdl:135]: This will work but is a bit dangerous if "POS" were to occur in any of the actual data lines. Could use instead:

`tail -n +2 "$file" >> "~{out_prefix}_~{sample_name}_gwas.tab"`

File: tr-gwas/tr_gwas.wdl [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    String sample_name = basename(samples,"_plink.txt")
+
+    command <<<
+
+        #check if cohort is EUR or non_AFR, exclude PC6
+        
+        echo ~{sample_name} | grep -q -E "EUR|NOT_AFR"
+        if [ $? -eq 0 ]; then
```
Comment (gymreklab) [inline tr-gwas/tr_gwas.wdl:101]: this will work but is pretty dangerous since (1) It's possible in the future PC6 might not be in column 10 and (2) I think that grep command might match anything with EUR or NOT_AFR in it, so we might make other cohorts in the future that have those strings in it where we didn't intend to get rid of this covariate and filter it by accident.

I think we need a separate PR to deal in more detail with covariates. We can merge this for now, but I would first add a "TODO" note here and also put a big warning message on the README that this manually filters PC6 for these cohorts and we need to update to do this in a more automated way.

In a future PR we should add a step that does this more robustly, probably needs to be a script that gets run before `run_tr_gwas` that does the following for the samples in the requested cohort: (1) looks at pairwise correlation of all covariates + the phenotype and removes highly correlated ones and (2) filtering for any categorical covariates as we do in Margoliash et al. e.g. statin usage and sex are examples of categorial covariates we have used.

File: tr-gwas/tr_gwas_aou.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+		cohort_array.append(path)
+	return cohort_array
+
+
+def main():
+	parser = argparse.ArgumentParser(__doc__)
+	parser.add_argument("--name", help="name of the run", required=True, type=str)
+	parser.add_argument("--ancestry-pred-path", help="Path to ancestry predictions",type=str, default="gs://fc-aou-datasets-controlled/v7/wgs/short_read/snpindel/aux/ancestry/ancestry_preds.tsv")
+	parser.add_argument("--dryrun", help="Don't actually run the workflow. Just set up", action="store_true")
+	parser.add_argument("--phenotype", help="name of the phenotype, seperated by comma", required=False, type=str)
```
Comment (gymreklab) [inline tr-gwas/tr_gwas_aou.py:63]: would rephrase help message as "comma-separated list of phenotypes to process. If not provided, process all available phenotypes"

**Review by gymreklab — CHANGES_REQUESTED** [review]
Thanks for submitting this PR!

I made mostly minor comments. Once those are addressed we can merge.

See in particular the comment on automatic filtering of covariates - this can be addressed in the next PR but should be high on our list to avoid accidental filtering of covariates we didn't mean to in future runs.

**Comment by nicholema** [comment]
Thank you for all the comments!  I made the changes suggested in the comments. If everything looks good I will confirm merge.
